[qa] Send segwit-encoded blocktxn messages in p2p-compactblocks

TheBlueMatt opened the pull request:
Test with segwit blocktxn messages, since that was missed in #8904, in accordance with the updates to the BIP at https://github.com/bitcoin/bips/pull/462.

TheBlueMatt committed:
[qa] Send segwit-encoded blocktxn messages in p2p-compactblocks

sdaftuar commented:
utACK, good catch on the missing test case.

If you add an additional commit that sets `block.nVersion=4` in the `build_block_on_tip()` function, then I believe this would address the outstanding issue in #8916: https://github.com/bitcoin/bitcoin/pull/8916#discussion_r83298887

TheBlueMatt commented:
Hmm, wasn't targeting this for backport, though I suppose it doesn't really matter... I'll add that commit.

On October 14, 2016 8:00:43 PM EDT, Suhas Daftuar <notifications@github.com> wrote:
>utACK, good catch on the missing test case.
>
>If you add an additional commit that sets `block.nVersion=4` in the
>`build_block_on_tip()` function, then I believe this would address the
>outstanding issue in #8916:
>https://github.com/bitcoin/bitcoin/pull/8916#discussion_r83298887
>
>-- 
>You are receiving this because you authored the thread.
>Reply to this email directly or view it on GitHub:
>https://github.com/bitcoin/bitcoin/pull/8922#issuecomment-253946959

TheBlueMatt committed:
[qa] Build v4 blocks in p2p-compactblocktests

This fixes an issue in backporting to 0.13 as 0.13 enforces SF
activation by block version lockin instead of through a hard-coded
block height.

TheBlueMatt commented:
@sdaftuar ok, that should do it, I believe...I suppose this should get a "Needs Backport" tag now.

btcdrak commented:
utACK a4ad37d4ef4bcd81bc9d867b277efdebc86bc2e8

MarcoFalke commented:
utACK a4ad37d

laanwj commented:
Backported in #8916
